packages/hebao_v2/contracts/base/libwallet/MetaTxLib.sol

```diff
@@ -86,12 +86,18 @@ library MetaTxLib
         returns (bool success)
     {
         require(msg.sender != address(this), "RECURSIVE_METATXS_DISALLOWED");
+
         require(metaTx.to == address(this));
-        require(metaTx.validUntil >= block.timestamp, "METATX_EXPIRED");
 
         uint gasLeft = gasleft();
         require(gasLeft >= (metaTx.gasLimit.mul(64) / 63), "OPERATOR_INSUFFICIENT_GAS");
 
+        // Update the nonce before the call to protect against reentrancy
+        require(isNonceValid(wallet, msg.sender, metaTx.nonce, metaTx.data.toBytes4(0)), "INVALID_NONCE");
+        if (metaTx.nonce != 0) {
+            wallet.nonce[msg.sender] = metaTx.nonce;
+        }
```

Comment:
If we try to protect users from themselves we also need to add this here when `metaTx.nonce == 0`:
```
} else {
    require(metaTx.requiresSuccess, "META_TX_WITHOUT_NONCE_REQUIRES_SUCCESS");
}
```
If `metaTx.requiresSuccess == false` a `nonce == 0` metatx can be reused as many times as the relayer wants, while still getting paid the fee.